Add documentation for landisgyr_heat_meter

vpathuis opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Add documentation for the new Landis+Gyr Heat Meter integration


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/73363
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/3457
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

vpathuis committed:
Add documentation for landisgyr_heat_meter

homeassistant commented:
Hi @pvianag, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#23084).

Once you do that we will be able to review and accept this pull request.

Thanks!

frenck commented:
Something seems to went wrong, could you please git rebase this PR? Thanks! 👍

vpathuis commented:
Mm, weird. I only did a sync fork on my fork so I could do some minor changes. I'll look into it.

klaasnicolaas commented:
Your branch is descended from the `current` branch, to solve this problem you must rebase your branch to the `next` branch. If this doesn't work, you can always start over from a new branch 😉

vpathuis committed:
Add explanation on how the conversion works.

vpathuis commented:
Hi @frenck, I think I fixed the problem. I also added a small commit. Do you think this is now ready to merge so it can be released together with the parent PR?

klaasnicolaas commented on source/_integrations/landisgyr_heat_meter.markdown:
```diff
@@ -0,0 +1,88 @@
+  - Sensor
+ha_iot_class: Local Polling
+ha_config_flow: true
+ha_release: 2022.7
```
```suggestion
ha_release: 2022.9
```

frenck commented on source/_integrations/landisgyr_heat_meter.markdown:
```diff
@@ -0,0 +1,88 @@
+
+The device is read through the optical interface. An (USB) IR reader is needed and connected to Home Assistant.
+
+<img src='/images/integrations/landisgyr_heat_meter/usb_ir_reader.png' />
```
Please use Markdown syntax to embed images.

frenck commented on source/_integrations/landisgyr_heat_meter.markdown:
```diff
@@ -0,0 +1,88 @@
+
+<img src='/images/integrations/landisgyr_heat_meter/usb_ir_reader.png' />
+
+<div class='note warning'>
```
Is this really a warning? This could also just be a normal text at the end of the sensors paragraph.

frenck commented on source/_integrations/landisgyr_heat_meter.markdown:
```diff
@@ -0,0 +1,88 @@
+
+The integration will create the following sensors:
+
+- heat usage (MWh)
+- volume usage (m3)
```
```suggestion
- Heat usage (MWh)
- Volume usage (m3)
```

frenck commented on source/_integrations/landisgyr_heat_meter.markdown:
```diff
@@ -0,0 +1,88 @@
+- heat usage (MWh)
+- volume usage (m3)
+
+To be compatible with the Home Assistant energy units of measurement, heat usage is converted to MWh, from usage in GJ which is supplied by the device, using conversion factor: 1 GJ = 0.277778 MWh.  
```
```suggestion
To be compatible with the Home Assistant energy units of measurement, heat usage is converted to MWh, from usage in GJ, which is supplied by the device, using a conversion factor: 1 GJ = 0.277778 MWh.
```

frenck commented on source/_integrations/landisgyr_heat_meter.markdown:
```diff
@@ -0,0 +1,88 @@
+
+To be compatible with the Home Assistant energy units of measurement, heat usage is converted to MWh, from usage in GJ which is supplied by the device, using conversion factor: 1 GJ = 0.277778 MWh.  
+
+Further data that is read from the device is added as diagnostic entities: 
```
```suggestion
Further data that is read from the device is added as diagnostic entities:
```

frenck commented on source/_integrations/landisgyr_heat_meter.markdown:
```diff
@@ -0,0 +1,88 @@
+
+Further data that is read from the device is added as diagnostic entities: 
+
+- heat usage measured in GJ, as is read from the device before conversion
+- ownership number
+- volume previous year (m3)
+- heat previous year (MWh)
+- error number
+- device number
+- measurement period minutes
+- power max kw
+- power max previous year kw
+- flow rate max m3ph
+- flow rate max previous year m3ph
+- flow temperature max c
+- return temperature max c
+- flow temperature max previous year c
+- return temperature max previous year c
+- operating hours
+- fault hours
+- fault hours previous year
+- yearly set day
+- monthly set day
+- meter date time
+- measuring range m3ph
+- settings and firmware
+- flow hours
```
```suggestion
- Heat usage measured in GJ, as is read from the device before conversion
- Ownership number
- Volume previous year (m3)
- Heat previous year (MWh)
- Error number
- Device number
- Measurement period minutes
- Power max kW
- Power max previous year kW
- Flow rate max m3ph
- Flow rate max previous year m3ph
- Flow temperature max c
- Return temperature max c
- Flow temperature max previous year c
- Return temperature max previous year c
- Operating hours
- Fault hours
- Fault hours previous year
- Yearly set day
- Monthly set day
- Meter date time
- Measuring range m3ph
- Settings and firmware
- Flow hours
```

frenck commented on source/_integrations/landisgyr_heat_meter.markdown:
```diff
@@ -0,0 +1,88 @@
+
+## Polling the device
+
+Polling is not automatic (see battery warning). This means that after adding the integration all values will initially be unknown.
+To read the values from the device, create an automation that will update one of the entities. The other entities will be updated as well.
```
That sounds off? We sound not make a broken sensor out of the box, not should we ever allow sensors that require manually created automations to function.

I would suggest restoring the state of this sensor (so we don't need to poll on startup) and poll, e.g., once a week or something like that.

vpathuis replied:
Agreed. This text was a leftover from an earlier version. State is now indeed restored on startup. It's also polled after adding the integration so the user should not see unknown values.

vpathuis replied:
Is that acceptable? I'd rather leave the polling to the user, but if it's not compliant with HA standards, of course I will change this.

frenck replied:
Well it doesn't violate any standards I think, but no everybody knows or uses YAML, confronting them with a YAML blob to get a working sensor, IMHO isn't the user experience we should or want to provide.

> State is now indeed restored on startup. It's also polled after adding the integration so the user should not see unknown values.

That means it updates once and never again? That is also weird. So it worked, but doesn't anymore (from a user perspective).

Alternatively, poll daily or weekly, and keep the entity disabled by default (and don't poll when it is). This allows the user to enable it if they want.

If they want another polling schedule, they can still opt to do that (e.g., do it weekly, monthly, or every minute).

vpathuis replied:
Thanks for the feedback. Disabling entities won't be necessary I think, because that would just disable all functionality and in that case the user could just remove the integration. 

Creating the automation would not require using YAML. The example is just shownn for convenience. I now understand it has the oposite effect :-o.

The idea behind the chosen solution is:
- the user probably only wants to poll daily as a maximum, due to the risk of battery drain (this could only be replaced by a specialist, so expensive). This is ultimately for the user to decide. 
- the user wants as much control on getting the most relevant detail as possible, within this limitation, so probably polling late in the day, or week, or perhaps before and after the weekend. Allowing the user to do it manually will give max control. 

I also looked at options to set a time in config flow, but I felt that was not the right place. 

What I could do is add a setting in the config flow to enable or disable polling? If enabled, it will poll once weekly (without time option). If disabled, the user can do it manually using automations. How does that sound? 

This will require changes to the PR that is already merged. How does that work, should I just add another PR?

frenck replied:
> Creating the automation would not require using YAML. The example is just shownn for convenience. I now understand it has the oposite effect :-o.

Things remain the same, if it is a polling integration, we need to provide a sane default and usable polling frequency. If one decides to change, that is up to them.

> What I could do is add a setting in the config flow to enable or disable polling?

That already exists. For every integration.

vpathuis replied:
> Things remain the same, if it is a polling integration, we need to provide a sane default and usable polling frequency. If one decides to change, that is up to them.

Alright, understood.
 
> That already exists. For every integration.

Oh yes I see now. That's very convenient! :)
I'll apply the change and then also change the documentation.

vpathuis replied:
I've submitted the change here: https://github.com/home-assistant/core/pull/77078
The integration will poll daily by default. It's what most people seem to do, so it's fine.
I've also changed the documentation accordingly. 
I look forward to your further suggestions or feedback!

vpathuis committed:
Apply suggested feedback on PR

vpathuis committed:
Update documentation to reflect new polling

frenck commented on source/_integrations/landisgyr_heat_meter.markdown:
```diff
@@ -0,0 +1,90 @@
+
+### Polling manually (optional)
+
+For detailed control on when the device is polled. disable default polling for this integration and create an automation that will update one of the entities. The other entities will be updated as well.
+
+If you're comfortable with yaml, this yaml could be used:
```
```suggestion
For detailed control on when the device is polled, disable the default polling for this integration and create an automation that will update one of the entities (the other entities will be updated as well)

If you're comfortable with YAML, this game could be used:

```

frenck commented on source/_integrations/landisgyr_heat_meter.markdown:
````diff
@@ -0,0 +1,90 @@
+
+If you're comfortable with yaml, this yaml could be used:
+```yaml
+alias: Heat Meter daily update
+description: ''
+trigger:
+  - platform: time
+    at: '23:30:00'
+condition: []
+action:
+  - service: homeassistant.update_entity
+    data: {}
+    target:
+      entity_id: sensor.heat_meter_heat_usage_gj
+mode: single
````
```suggestion
alias: "Heat Meter manual update"
trigger:
  - platform: time
    at: "23:30:00"
action:
  - service: homeassistant.update_entity
    target:
      entity_id: sensor.heat_meter_heat_usage_gj
```

frenck committed:
Simplify example

frenck committed:
Tiny lang tweak

frenck reviewed (approved):
Thanks, @vpathuis 👍
